### PR TITLE
Change SigFlags into an enum.

### PR DESCRIPTION
**Pull request by chaosagent**
Addresses #459.

This is a breaking change.

Should SigFlags be renamed to something more sensible?

### CODE REVIEW

**Comment by fiveop** [comment]
The PR looks good.

I agree that the name should be different. I cannot think of a good naming convention that would work for all enumerations (e.g. `SigEnum` just does not tell us anything). Therefore, I would propose to go with `SigHow`.

In addition, could you please add an entry to CHANGELOG.md describing the breaking change.

**Comment by chaosagent** [comment]
I was thinking of something like `SigmaskHow` since this enum only applies to to the `pthread_sigmask` and the `sigmask` (unimplemented in nix) calls but not the rest of the signal functions. In contrast, `SigSet` applies to both `sigaction` and `sigmask`, so it deserves to be labelled generically as `Sig`. Another possibility is to just use `Request` as the standard way to specify `how`-like values in the library like with PtraceRequest, which is actually enumerated in C `libc` [1], but using `How` might be advantageous in that it reflects the argument names in `libc`. Thoughts?

I'm probably making way too much of a fuss about naming though ;)

I'll hold off on the CHANGELOG changes until we figure this out I guess.
[1] http://man7.org/linux/man-pages/man2/ptrace.2.html

**Comment by chaosagent** [comment]
I went ahead and pushed a rename to `SigmaskHow` and an entry in `CHANGELOG.md`.

File: CHANGELOG.md [hunk 1]
```diff
@@ -36,6 +36,9 @@ This project adheres to [Semantic Versioning](http://semver.org/).
   accessible with the new method `events()` of `EpollEvent`. Instances of
   `EpollEvent` can be constructed using the new method `new()` of EpollEvent.
   ([#410](https://github.com/nix-rust/nix/pull/410))
+- `SigFlags` in `::nix::sys::signal` has be renamed to `SigmaskHow` and its type
+  has changed from `bitflags` to `enum` in order to conform to our conventions.
+  ([#410](https://github.com/nix-rust/nix/pull/460))
```
Comment (fiveop) [inline CHANGELOG.md]: Link name is #410 instead of #460.

**Comment by fiveop** [comment]
@homu r+

**Pull request merged**